### ISSUE TITLE
Bump Filament 1.71.5 → 1.71.6 to fix DisplayHelper NPE race (google/filament#9352)

### ISSUE THREAD

**Issue by michihiro-ikeda-safie** [sceneview/sceneview#2708]
## Is your feature request related to a problem?

Yes — a production crash. Filament ≤ 1.71.5 has a race condition in
`DisplayHelper` that causes a `NullPointerException` when a display-changed
event (e.g. an adaptive refresh-rate switch) is delivered right after
`DisplayHelper.detach()` has nulled `mRenderer`:

```
java.lang.NullPointerException: Attempt to invoke virtual method
'com.google.android.filament.Renderer$DisplayInfo com.google.android.filament.Renderer.getDisplayInfo()'
on a null object reference
    at com.google.android.filament.android.DisplayHelper.updateDisplayInfo(DisplayHelper.java:136)
    at com.google.android.filament.android.DisplayHelper$1.onDisplayChanged(DisplayHelper.java:102)
    at android.hardware.display.DisplayManagerGlobal$DisplayListenerDelegate.handleDisplayEventInner(...)
```

This was reported upstream as google/filament#9352 and fixed in
google/filament@180e91f5 ("Fix race conditions in DisplayHelper.java (#10025)"),
first shipped in **Filament v1.71.6** (2026-06-10). SceneView 4.x currently pins
1.71.5, so every SceneView release ships the unfixed code.

In our app this crash spiked ~30x/day after migrating SceneView 2.3.0 → 4.17.0.
The reason is a teardown ordering change: in 2.3.0 `onDetachedFromSurface` ran
`displayHelper.detach()` **before** `destroySwapChain` + `engine.flushAndWait()`,
while 4.x `SceneRenderer.onDetachedFromSurface` runs `flushAndWait()` **before**
`detach()`. `flushAndWait()` blocks the main thread, so a refresh-rate-change
event (triggered by removing the video surface — very common on Android 16
devices with adaptive refresh; ~90% of the crashes we observe are on Android 16)
gets queued during the flush and is delivered after `detach()` nulls `mRenderer`.
Filament 1.71.6's lock + stale-state guard fixes this.

## Describe the solution you'd like

Bump `filament = "1.71.5"` → `"1.71.6"` in `gradle/libs.versions.toml`.

## Describe alternatives you've considered

- Additionally/alternatively, reordering `SceneRenderer.onDetachedFromSurface`
  to call `displayHelper?.detach()` first (as 2.3.0 did) would shrink the race
  window even on older Filament versions.
